data_steward/deid/aou.py

```diff
@@ -298,24 +294,37 @@ def initialize(self, **args):
             self.map_questionnaire_response_ids(million)
             create_person_id_src_hpo_map(self.idataset, self.credentials)
 
-        if person_table.shape[0] > 0:
-            person_table = person_table[person_table.age < age_limit]
-            map_sql = 'SELECT * FROM ' + map_tablename
-            job_config = {
-                'query': {
-                    'defaultDataset': {
-                        'datasetId': self.idataset
-                    }
-                }
-            }
-            map_table = self.get_dataframe(sql=map_sql, query_config=job_config)
-            if not map_table.shape[0] > 0:
-                LOGGER.error(
-                    "Unable to initialize Deid. _deid_map has no participants to load."
-                )
+        # ensure mapping table only contains participants within age limits
+        sql = (f"SELECT DISTINCT p.person_id, "
+               f"EXTRACT(YEAR FROM CURRENT_DATE()) - year_of_birth AS age "
+               f"FROM {self.idataset}.person AS p "
+               f"JOIN {map_tablename} AS map "
+               f"USING (person_id) "
+               f"ORDER BY age")
+        job_config = {'query': {'defaultDataset': {'datasetId': self.idataset}}}
+        person_table = self.get_dataframe(sql=sql, query_config=job_config)
+        LOGGER.info(f"possible patient count is:\t{person_table.shape[0]}")
+
+        # ensure age eligible participants exist in the mapping table
+        eligible_person_table = person_table[person_table.age < age_limit]
+        if eligible_person_table.shape[0] < 1:
+            LOGGER.error(
+                f"Unable to initialize Deid. {map_tablename} table cannot be "
+                f"joined to {self.idataset}.person table to verify age requirements."
+            )
+
+        # ensure no age ineligible participants are available in the mapping table
+        ineligible_person_table = person_table[person_table.age >= age_limit]
+        if ineligible_person_table.shape[0] > 0:
+            LOGGER.error(f"{ineligible_person_table.shape[0]} age ineligible "
+                         f"participants are available in "
+                         f"{map_tablename}.  Deid is bailing out!!")
+
+        LOGGER.info(f"map table contains {eligible_person_table.shape[0]} "
+                    f"records.")
```

Comment:
nit: This can be removed since it is the same as above

Reply:
Which line is it duplicating?